pkg/task/task.go

```diff
@@ -165,7 +166,14 @@ func DefaultStartTask(conf *Config) {
 			startTime := time.Now()
 			err := conf.Func.Execute(conf.Context, logger)
 			executionDuration := time.Since(startTime)
-			if err != nil && err != context.Canceled {
+			switch {
+			case err == nil:
+			case errors.Is(err, context.Canceled),
+				errors.Is(err, errors.ErrContextCancelled),
+				errors.Is(err, context.DeadlineExceeded),
+				errors.Is(err, errors.ErrContextDeadlineExceeded):
+			case errors.Is(err, io.EOF):
+			default:
```

Comment:
This magic is already supported by `pkg/errors`: https://github.com/TheThingsNetwork/lorawan-stack/blob/v3.23/pkg/errors/codes.go#L45-L50

This is cleaner than testing error types as what you really want is knowing whether the error is canceled or a deadline has exceeded, i.e. the code.

Note also that we use US English "canceled" with one l.

Reply:
Okay please decorate this with a comment then.